mariadb/templates/pvc.yaml

```diff
@@ -0,0 +1,15 @@
+{{- if .Values.usePersistence }}
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: {{ template "fullname" . }}
+  annotations:
+    volume.alpha.kubernetes.io/storage-class: {{ .Values.dataVolume.storageClass | quote }}
```

Comment:
Here is how it works (@saad-ali can you proofread what I wrote here):

- if you put `volume.alpha.kubernetes.io/storage-class = X` annotation key on your PVC, it does not matter what the value of X is.  It always get the default storage class. (same behavior on 1.3 or 1.4)
- if you are running 1.4, and you have not created a default storageClass object, and you do not have the alpha label on your PVC, then nothing happens.
- if you are running 1.4 and you have previously created a default storageClass object on your cluster, then when the PVC is created, the `volume.beta.kubernetes.io/storage-class = Y` annotation will magically be added to your PVC when you create it.   Here Y is the name of your default storage class object.  Y may differ across cloud providers, but you don't need to know it in the chart.
- if you are using 1.4, and you have a non-default storageClass object called `Z`, and you put  the annotation ``volume.beta.kubernetes.io/storage-class = Z` on your PVC, it gets a type `Z` PV automatically.